src/Vault.ts

```diff
@@ -136,7 +142,12 @@ export class Vault {
             qs: parameters,
         };
 
-        const res = await request(requestOptions);
+        let res = await request(requestOptions);
+
+        if (retryWithTokenRenew && res.statusCode === 403) {
```

Comment:
If we're enabling auto-renewal per request, do we still need the background timer at all?